### PR TITLE
specify inbound protocol for sandwiched waypoint

### PR DESCRIPTION
**Pull request by stevenctl**
`ambient.istio.io/waypoint-inbound-binding` is an interim API for telling the zTunnel _how_ to sandwich a waypoint. This annotation lives on the `GatewayClass` in place of the "gateway capabilities" suggestion from the [waypoint capture doc](https://docs.google.com/document/d/1DAM_6SajCyC2hD8wBUraaZWZJqUVYPyemzw0-xpHmdY/edit). In the future this may be specified using ParametersRef. 

Sample usage:

https://gist.github.com/stevenctl/ec2ddae79734f8f9367400961ca9c48f

### CODE REVIEW

**Review by bleggett — APPROVED** [review]

File: pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go [hunk 1]
```diff
@@ -279,10 +279,21 @@ func (a *index) podWorkloadBuilder(
...
 		}
 
+		if instancedWaypoint := fetchWaypointForInstance(ctx, Waypoints, p.ObjectMeta); instancedWaypoint != nil {
+			// we're an instance of a waypoint, set inbound tunnel info
+			w.ApplicationTunnel = &workloadapi.ApplicationTunnel{
+				Protocol: instancedWaypoint.DefaultBinding.Protocol,
+				Port:     instancedWaypoint.DefaultBinding.Port,
+			}
+		} else if waypoint := fetchWaypointForWorkload(ctx, Waypoints, Namespaces, p.ObjectMeta); waypoint != nil {
+			println("stevenctl ", p.Name, " has a  waypoint ")
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go:293]: (and some commented code above)

**Review by howardjohn — COMMENTED** [review]
overall lgtm, just needs minor cleanup

**Comment by stevenctl** [comment]
@bleggett / @howardjohn  WDYT about the ServiceEntry/WorkloadEntry codepaths? I _could_ imagine those pointing to some external Waypoint. In that case, they probably won't have the label. Instead I was thinking we check if their address is in `Waypoint.Addresses` to see if they're an instance. WDYT?

**Review by howardjohn — APPROVED** [review]

**Comment by bleggett** [comment]
> @bleggett / @howardjohn about the ServiceEntry/WorkloadEntry codepaths.. I could imagine those pointing to some external Waypoint. In that case, they probably won't have the label. Instead I was thinking we check if their address is in `Waypoint.Addresses` to see if they're an instance. WDYT?


Sorry, `external` as in the gatewayclass is not actually visible to the control plane?

Or do you just mean a managed waypoint indirected by a SE? 

If latter, I think we probably do need to do a deeper check.

**Comment by stevenctl** [comment]
You could have a GatewayClass `foo` used by a Gateway `my-external-gateway` with address `1.2.3.4`. If we see some entry with `1.2.3.4`, we might say it's a "gateway instance". This is less important for adding the inbound protocol stuff, more important to avoid setting `waypoint` on the workload for the waypoint itself.

**Pull request merged**